### PR TITLE
fix fail_with errors

### PR DESCRIPTION
**Pull request by firefart**
pre work for #5171 

cc @wvu-r7 @wchen-r7

### CODE REVIEW

**Comment by FireFart** [comment]
all occurences fixed

File: modules/exploits/linux/smtp/exim_gethostbyname_bof.rb [hunk 1]
```diff
@@ -543,10 +543,10 @@ def try_code_execution(len, what, where)
   DOT = '[.]'
 
   def smtp_connect(exploiting = true)
-    fail_with("smtp_connect", "sock isn't nil") if sock
+    fail_with(Failure::Unknown, "sock isn't nil") if sock
 
     connect
-    fail_with("smtp_connect", "sock is nil") if not sock
+    fail_with(Failure::Unkown, "sock is nil") if not sock
```
Comment (wvu-r7) [inline modules/exploits/linux/smtp/exim_gethostbyname_bof.rb]: ```Failure::Unknown```

Comment (FireFart) [inline modules/exploits/linux/smtp/exim_gethostbyname_bof.rb]: fixed

**Comment by wvu-r7** [comment]
All changes valid now. Thanks.

**Pull request merged**